src/hooks/ProjectENSNameResolver.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react'
+
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { readProvider } from 'constants/readProvider'
+
+const JUICEBOX_V2_TXT_RECORD_KEY = 'juicebox_v2'
```

Comment:
We'll need some docs (or potentially UI instruction) for setting this TXT record in ENS

Reply:
@aeolianeth will it be possible to do programatically?

Reply:
yeah i think it actually might be. we might be able to queue up the TX to set the text record on our end. would be dope

Reply:
Indeed. Even a modal to buy an ENS name if they don't have one yet 👀

Reply:
> Also kind of a side note, could we just use "juicebox" for the record name? There's a UX element here for anyone who chooses to update the record manually, and adding the _v2 may seem a little awkward considering we don't use this for v1 (or if you don't even know about jb versions)

yeah good call, I was thinking that somehow we could extend this to support V1 and V2. Probably out of scope though.

Renamed it to `juicebox` 👍